src/main/java/adam/task/Task.java

```diff
@@ -0,0 +1,65 @@
+package ada.task;
+
+/**
+ * Represents a task to be stored in the list of tasks.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Returns a task containing its description and is marked by default as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns if the task is done or not done.
+     *
+     * @return "X" if the task is done, " " if not done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void mark() {
+        this.isDone = true;

```

Comment:
Its a good practice to omit the `this` keyword when accessing class members and you should only use it when there are ambiguous variable names in the scope (i.e. in constructors or when your method arguments have the same 
name)
```suggestion
        isDone = true;
```